src/Take.Blip.Builder/LazyInput.cs

```diff
@@ -47,7 +47,9 @@ public class LazyInput
                     return await artificialIntelligenceExtension.AnalyzeAsync(
                         new AnalysisRequest
                         {
-                            Text = _lazySerializedContent.Value
+                            Text = _lazySerializedContent.Value,
+                            MessageId = Message.Id,
+                            UserIdentity = Message.From.ToIdentity().ToString(),
```

Comment:
Receive the `userIdentity` in the LazyInput constructor and passes the same value that is passed to the IContextProvider here:
https://github.com/takenet/blip-sdk-csharp/blob/master/src/Take.Blip.Builder/FlowManager.cs#L150

The userIdentity in some cases may be different than the message.From value.